src/Tickets/Commerce/Gateways/Stripe/Hooks.php

```diff
@@ -53,6 +59,57 @@ protected function add_filters() {
 		add_filter( 'tec_tickets_commerce_admin_notices', [ $this, 'filter_admin_notices' ] );
 	}
 
+	/**
+	 * Set up Stripe Webhook based on transient value.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function setup_stripe_webhook_on_activation() {
+		/**
+		 * Filters whether to enable the Stripe Webhook.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool $need_to_enable_stripe_webhook Whether to enable the Stripe Webhook.
+		 */
+		$need_to_enable_stripe_webhook = apply_filters( 'tec_tickets_commerce_need_to_enable_stripe_webhook', get_transient( 'tec_tickets_commerce_setup_stripe_webhook' ) );
+
+		if ( false === $need_to_enable_stripe_webhook ) {
+			return false;
+		}
+
+		// Always delete the transient.
+		delete_transient( 'tec_tickets_commerce_setup_stripe_webhook' );
+
+		// Bail in non truthy values as well.
+		if ( ! tribe_is_truthy( $need_to_enable_stripe_webhook ) ) {
```

Comment:
Yes i had a reason in mind. The reason is how get_transient is designed to work. If the stored transient does not exist or is expired it will return (bool) false. As a result we don't need to delete_transient in this case.

My though was that if 3rd party wants to use the filter they can `return 0;` or they can `return false;` and take care of deleting the transient as well